tools/proto2jsonschema/gen.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+set -e
+
+# Only run in repo root
+currentDir=${PWD##*/}
+
+if [ "$currentDir" != "google-cloudevents" ] ; then
+  echo "Error: Script must be run in google-cloudevents root.";
+  exit 1;
+fi
+
+# Install protoc
+case "$OSTYPE" in
+  linux*)
+    PROTOBUF_PLATFORM=linux-x86_64
+    PROTOC=tmp/bin/protoc
+    ;;
+  win* | msys* | cygwin*)
+    PROTOBUF_PLATFORM=win64
+    PROTOC=tmp/bin/protoc.exe
+    ;;
+  darwin*)
+    PROTOBUF_PLATFORM=osx-x86_64
+    PROTOC=tmp/bin/protoc
+    ;;
+  *)
+    echo "Unknown OSTYPE: $OSTYPE"
+    exit 1
+esac
+
+rm -rf tmp
+mkdir tmp
+cd tmp
+echo "- Downloading protobuf tools"
+PROTOBUF_VERSION=3.12.3
+curl -sSL \
+  https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOBUF_VERSION/protoc-$PROTOBUF_VERSION-$PROTOBUF_PLATFORM.zip \
+  --output protobuf.zip
+unzip -q protobuf.zip
+cd ..
+
+echo "- Setting up protoc plugin (chrusty/protoc-gen-jsonschema)"
+go get -v github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
+go install github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
+
+echo "- Converting protos to JSON Schemas"
+chmod +x $PROTOC
+
+# Clone googleapis dependency
+cd proto
+if [ ! -d "googleapis" ] ; then
+  git clone https://github.com/googleapis/googleapis
```

Comment:
This is cloning in the same place you've got a submodule - that would leave the repository in an inconsistent state.
Again, I don't think we need the submodule at all, but if we *do* want it, this script should expect it to be there as a submodule. If it's not there, error out with a message about using `git submodule init` etc.

Reply:
There should be no cloning of googleapis, this was never used as part of the `protoc` generation process and was a mistake adding it. This instruction is outdated.